server/src/internalClusterTest/java/org/elasticsearch/health/GetHealthActionIT.java

```diff
@@ -8,93 +8,134 @@
-        HealthIndicatorResult nodeDoesNotHaveMaster = controller.indicators().get(ClusterCoordination.INSTANCE_HAS_MASTER_NAME);
-        assertEquals(ClusterCoordination.INSTANCE_HAS_MASTER_NAME, nodeDoesNotHaveMaster.name());
-        assertEquals(HealthStatus.GREEN, nodeDoesNotHaveMaster.status());
-        assertEquals(ClusterCoordination.INSTANCE_HAS_MASTER_GREEN_SUMMARY, nodeDoesNotHaveMaster.summary());
+        @Override
+        public HealthIndicatorResult calculate() {
+            var status = clusterService.getClusterSettings().get(TEST_HEALTH_STATUS);
+            return createIndicator(status, "Health is set to [" + status + "] by test plugin", HealthIndicatorDetails.EMPTY);
+        }
     }
 
-    public void testGetHealthInstanceNoMaster() throws Exception {
-        // builds the coordinating-only client before disrupting all nodes
-        final Client client = internalCluster().coordOnlyNodeClient();
-
-        final NetworkDisruption disruptionScheme = new NetworkDisruption(
-            new NetworkDisruption.IsolateAllNodes(new HashSet<>(Arrays.asList(internalCluster().getNodeNames()))),
-            NetworkDisruption.DISCONNECT
-        );
-
-        internalCluster().setDisruptionScheme(disruptionScheme);
-        disruptionScheme.startDisrupting();
-
-        try {
-            assertBusy(() -> {
-                ClusterState state = client.admin().cluster().prepareState().setLocal(true).execute().actionGet().getState();
-                assertTrue(state.blocks().hasGlobalBlockWithId(NoMasterBlockService.NO_MASTER_BLOCK_ID));
-
-                GetHealthAction.Response response = client.execute(GetHealthAction.INSTANCE, new GetHealthAction.Request()).get();
-                assertEquals(HealthStatus.RED, response.getStatus());
-                assertEquals(2, response.getComponents().size());
-                HealthComponentResult controller = response.getComponents()
-                    .stream()
-                    .filter(c -> c.name().equals("cluster_coordination"))
-                    .findAny()
-                    .orElseThrow();
-                assertEquals(1, controller.indicators().size());
-                HealthIndicatorResult instanceHasMaster = controller.indicators().get(ClusterCoordination.INSTANCE_HAS_MASTER_NAME);
-                assertEquals(ClusterCoordination.INSTANCE_HAS_MASTER_NAME, instanceHasMaster.name());
-                assertEquals(HealthStatus.RED, instanceHasMaster.status());
-                assertEquals(ClusterCoordination.INSTANCE_HAS_MASTER_RED_SUMMARY, instanceHasMaster.summary());
-            });
-        } finally {
-            internalCluster().clearDisruptionScheme(true);
+    public void testGetHealth() throws Exception {
+
+        var client = client();
+        var status = randomFrom(HealthStatus.values());
+
+        try (var ignored = setTestSetting(TEST_HEALTH_STATUS, (builder, setting) -> builder.put(setting.getKey(), status))) {
```

Comment:
I think I would prefer to avoid the `setTestSetting` method and just use `setClusterSettings(Setttings.builder().put(TEST_HEALTH_STATUS, status)`, that seems simpler and avoids this extra test framework method?

We can avoid the reset by making the scope of the integration test `TEST` rather than `SUITE`, given that we mess with cluster settings that seems more appropriate.